### DISCUSSION TITLE
SST perturbation experiment

### DISCUSSION THREAD

**Discussion by PappuP** [E3SM-Project/E3SM#6612] [E3SM model help]
Hello, I am currently planning to run SST perturbation experiment (--compset F2010 --res ne30pg2_r05_IcoswISC30E3r5). So my plan is to modify the default (1 degree) SST input file in different location such as West Pacific , East Pacific, Southern Ocean, and many other location. 

For each of the modification do I need to create new case & build every time? As The only thing I want to change the path of **SSTICE_DATA_FILENAME**. I already create, build and run West Pacific (WP) experiment. **Can I do create_clone from WP case instead of creating new case and build for all the experiment?** which saves a lot of time in my case as I do have a lot of different SST modification. Or hybrid run or branch run would be better?

My understanding is that model use SST data only when it runs? So it is time-consuming and unnecessary as the same executable may be used for multiple SST experiment?

**Comment by mahf708** [comment]
Yes, the SST setting in question is a runtime parameter. Would you be willing to share your workflow script(s) with us to help you further?

I think you are most concerned about the build process, correct? Creating new cases (without building them) is relatively painless.

If I were you, I would follow the following workflow. I would create a baseline case and build it as normal; if desired, run it. Let's assume this case has the path `/path/to/my/base/case/`. Then, I would set up new cases (the sensitivity cases you're looking to conduct) where I would set `do_case_build=false` ([here](https://github.com/E3SM-Project/E3SM/blob/f167567e6f8de86f95b8ce59a89ba11f146ff0e0/run_e3sm.template.sh#L106)) and set the `readonly OLD_EXECUTABLE=/path/to/my/base/case/build/e3sm.exe` ([here](https://github.com/E3SM-Project/E3SM/blob/f167567e6f8de86f95b8ce59a89ba11f146ff0e0/run_e3sm.template.sh#L100)). It may help to use a script that is similar to the one we provide in the root of the repo: https://github.com/E3SM-Project/E3SM/blob/master/run_e3sm.template.sh. Also be sure to include other edits as desired in your sensitivity runs. The namelist setting `SSTICE_DATA_FILENAME` is the one you likely have in mind.

Let us know if this helps, and best of luck!

**Reply by PappuP** [reply]
I am using the same script as you directed. Thats a nice suggestion I will follow this. **Another thing is I want to start all the run from the ending of my baseline case. What modification I need to do in the script for it?**

**Reply by mahf708** [reply]
You will need to utilize [these lines](https://github.com/E3SM-Project/E3SM/blob/f167567e6f8de86f95b8ce59a89ba11f146ff0e0/run_e3sm.template.sh#L40-L44) from the script.

I recommend the following. 

First, make sure your run ends at a _restart_ point (this is controlled by these settings in the run script):

```sh
  # Production simulation
  readonly CASE_SCRIPTS_DIR=${CASE_ROOT}/case_scripts
  readonly CASE_RUN_DIR=${CASE_ROOT}/run
  readonly PELAYOUT="L"
  readonly WALLTIME="34:00:00"
  readonly STOP_OPTION="nyears"
  readonly STOP_N="50"
  readonly REST_OPTION="nyears"
  readonly REST_N="5"
  readonly RESUBMIT="9"
  readonly DO_SHORT_TERM_ARCHIVING=false
```

In particular, one way to ensure it is to make STOP_OPTION and STOP_N be identical to REST_OPTION and REST_N respectively. 

Second, in the baseline case, utilize the short-term archive functionality to "short-term" archive the case. Simply `cd /path/to/my/base/case_name/case_scripts` and then `./case.st_archive`. Once, done, you should be able to see directories inside `/path/to/my/base/case/archive` such as `rest`.

Third, you could edit these lines such that the `RUN_REF*` entries correspond to the end of your baseline run (case_name):

```sh
# Run options
readonly MODEL_START_TYPE="branch"  # 'initial', 'continue', 'branch', 'hybrid'
readonly START_DATE="2020-01-01"

# Additional options for 'branch' and 'hybrid'
readonly GET_REFCASE=TRUE
readonly RUN_REFDIR="/path/to/my/base/case_name/archive/rest/2020-01-01-00000"
readonly RUN_REFCASE="case_name"
readonly RUN_REFDATE="2020-01-01"   # same as MODEL_START_DATE for 'branch', can be different for 'hybrid'
```

**Reply by mahf708** [reply]
Let us know if it works for you, and if so, please mark the response as answered :)

**Reply by PappuP** [reply]
Hi I am little bit confused for the third step, I used F2010 compset and if I run my baseline case for 21 years. **What should be my START_DATE and RUN_REFDATE in this part of the script for other cases?**
```````````
readonly START_DATE="?"
# Additional options for 'branch' and 'hybrid'
readonly GET_REFCASE=TRUE
readonly RUN_REFDIR="/path/to/my/base/case_name/archive/rest/2020-01-01-00000"
readonly RUN_REFCASE="case_name"
readonly RUN_REFDATE="?"   # same as MODEL_START_DATE for 'branch', can be different for 'hybrid'
`````````

**Reply by mahf708** [reply]
What are your MODEL_START_TYPE, START_DATE, STOP_OPTION, STOP_N, REST_OPTION, and REST_N in the baseline case?

Generally, for "branch" run, the date will be the one that you will see in archive/rest. If you started your F2010 case from 0001-01-01, that will be 0022-01-01 (after 21 years).

**Reply by PappuP** [reply]
````
readonly MODEL_START_TYPE="initial"  # 'initial', 'continue', 'branch', 'hybrid'
readonly START_DATE="0001-01-01"
  readonly STOP_OPTION="nyears"
  readonly STOP_N="1"
  readonly REST_OPTION="nyears"
  readonly REST_N="1"
  readonly RESUBMIT="20"
  readonly DO_SHORT_TERM_ARCHIVING=false

**Reply by mahf708** [reply]
Yeah, so it should be 0022-01-01. Please remember to do `./case.st_archive` for the above to work. You should be able to see the restart dates under archive/rest and then you know you have the right numbers, etc.

**Reply by PappuP** [reply]
Great, thank you so much, I will try it when my baseline case run end. Then I will mark it as answered
